### ISSUE TITLE
Flutter Inspector is gone after 59.0.4

### ISSUE THREAD

**Issue by shinayser** [flutter/flutter-intellij#5694]
Just upgraded to 59.0.4 and my flutter inspector is gone from the lateral tab.
Where do I find it now?

I am using Intellij Community 2021.2

![image](https://user-images.githubusercontent.com/1146982/127920496-2a54eab2-5c53-47bf-ae08-8e5d3b95081e.png)

**Comment by josh-burton** [comment]
It seems this must be related to `Skip Flutter tool windows for non-Flutter projects (WIP) (#5622)`.

I've tried to find a way to configure my project as a Flutter project by adding a Flutter facet but it's not available as an option.

**Comment by josh-burton** [comment]
I was able to work around this by going to `Project settings -> Modules` and manually adding my modules by using the `+ -> Import Module`. 

I think this should work out of the box though.

**Comment by helin24** [comment]
@josh-burton that's what I was about to suggest, but I agree it would be better if we checked for these projects initially.

@shinayser are you also in a situation where your flutter project is below the top-level project that you have open?

**Comment by shinayser** [comment]
> @josh-burton that's what I was about to suggest, but I agree it would be better if we checked for these projects initially.
> 
> @shinayser are you also in a situation where your flutter project is below the top-level project that you have open?

Yes, I am using monorepo architecture.

**Comment by josh-burton** [comment]
I think another option would be to just enable dart support on the top level folder/module in project settings (I haven't tested this):

<img width="918" alt="Screen Shot 2021-08-03 at 9 39 29 AM" src="https://user-images.githubusercontent.com/848767/127927464-f60bbaff-713a-41ea-83b9-fdd4808b83c7.png">

**Comment by shinayser** [comment]
> I think another option would be to just enable dart support on the top level folder/module in project settings (I haven't tested this):
> 

Just tested this one withot success.

**Comment by stevemessick** [comment]
The Flutter plugin requires a `pubspec.yaml` in the top-level project directory. We have an [issue](https://github.com/flutter/flutter-intellij/issues/5291) to change this, but until it is addressed you are going to find various problems working with this repository structure.

**Comment by shinayser** [comment]
> The Flutter plugin requires a `pubspec.yaml` in the top-level project directory. We have an [issue](https://github.com/flutter/flutter-intellij/issues/5291) to change this, but until it is addressed you are going to find various problems working with this repository structure.

The plug-in was working fine until this morning, when I updated to 59.0.4

**Comment by AlexV525** [comment]
https://user-images.githubusercontent.com/15884415/127946595-54d7b2e2-e350-4c53-9b4b-d3fb9fe648c6.mp4

Facing the same issue, some projects are unable to activate the inspector window after upgraded the plugin from 58.0.4 to 59.0.4.

**Comment by josh-burton** [comment]
> The Flutter plugin requires a `pubspec.yaml` in the top-level project directory. We have an [issue](https://github.com/flutter/flutter-intellij/issues/5291) to change this, but until it is addressed you are going to find various problems working with this repository structure.

Seems like this change should be reverted until others areas of the plugin are ready for a mono repo structure.

**Comment by stevemessick** [comment]
@shinayser 
> The plug-in was working "ok" until this morning, when I updated to 59.0.4

That statement puzzles me. If you opened a project which contains a Flutter project, none of the Flutter-specific functionality should be available. Had you done something to configure it to sort-of work? I'm sure that not everything worked, but maybe enough was available to look like it worked.

**Comment by shinayser** [comment]
> @shinayser
> 
> > The plug-in was working "ok" until this morning, when I updated to 59.0.4
> 
> That statement puzzles me. If you opened a project which contains a Flutter project, none of the Flutter-specific functionality should be available. Had you done something to configure it to sort-of work? I'm sure that not everything worked, but maybe enough was available to look like it worked.

I mean, the side "Flutter inspectors" bars were working fine. They were all appearing and workign as properly.

**Comment by stevemessick** [comment]
Well, it seems you were able to use some parts of the Flutter plugin, but a number of UI features were already missing for you.

Let me rephrase my original comment. We currently only support Flutter projects that are opened as top-level projects.

I tend to agree, though, that this sort of complex project ought to be supported. I think we may need to change some plans to adapt. @devoncarew for consideration.

**Comment by AlexV525** [comment]
> Facing the same issue, some projects are unable to activate the inspector window after upgraded the plugin from 58.0.4 to 59.0.4. Also no logs for IDEA were produced in the console.

The inspector keep disappeared and can be only shown after I deleted the `.idea`.

**Comment by mono0926** [comment]
I was able to work around this by adding this to the top-level project directory

```yaml
# https://github.com/flutter/flutter-intellij/issues/5694
name: dummy
publish_to: 'none'
dependencies:
  flutter:
    sdk: flutter
```

**Comment by yimao009** [comment]
https://github.com/flutter/flutter-intellij/issues/5388
In `Project Settings`  clicked `Add Content Root`
